### PR TITLE
RA-1859:Registration Summary:Cannot delete HSU  optional identifier

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RA-1859 

cc @dkayiwa @mks-d

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/registrationapp/form/RegisterPatientFormBuilder.java [hunk 1]
```diff
@@ -264,21 +264,25 @@ public static void resolvePatientIdentifierFields(NavigableFormStructure form, P
...
+			
 							if (identifierType  != null) {
 
 								// see if there is existing identifier with this value, if so, no need to update
 								for (PatientIdentifier oldIdentifier : patient.getPatientIdentifiers(identifierType)) {
-									if (oldIdentifier.getIdentifier().equals(parameterValue)) {
+									if (oldIdentifier.getIdentifier().equals(parameterValue)) {					
 										return;
+									} else if(oldIdentifier.getIdentifier().isEmpty()) {
+									   log.warn("you have an empty identifier fied");
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/registrationapp/form/RegisterPatientFormBuilder.java:277]: @sherrif10 ,thanks for this,you can change fied to field around this line

Comment (sherrif10) [reply omod/src/main/java/org/openmrs/module/registrationapp/form/RegisterPatientFormBuilder.java:277]: Thanks @HerbertYiga will resolve that

**Comment by sherrif10** [comment]
This is the screen shot that voids us not to save an empty identifier field

identifierField

**Comment by dkayiwa** [comment]
@sherrif10 can you take a second look at the instructions that i gave you on talk in regards to what you need to do?

**Comment by sherrif10** [comment]
ok thanks ,Let me take another look

**Comment by sherrif10** [comment]
Sorry @dkayiwa  for late response , i guess this should behave the same with my first work, if identifier is deleted, then no identifier should be returned

**Comment by dkayiwa** [comment]
@sherrif10 should i make a commit for what i mean?

**Comment by sherrif10** [comment]
cc @dkayiwa : As you had suggested , i tried to void the existing identifier then later remove method

**Comment by dkayiwa** [comment]
Committed at https://github.com/openmrs/openmrs-module-registrationapp/commit/312137cdf9bb2066d2dd3f1b9dea2d5c387ba44a

**Pull request closed without merging**